### PR TITLE
Clean-up the ChainSyncServers creation after recent modification to resource management

### PR DESCRIPTION
**Pull request by EncodePanda**
Please read through commit messages to understand the intent of this PR.

### CODE REVIEW

**Comment by EncodePanda** [comment]
@karknu I've added you as a reviewer because you were doing the recent changes around the chainSyncServers

**Review by karknu — APPROVED** [review]
LGTM.

File: ouroboros-consensus/src/Ouroboros/Consensus/MiniProtocol/ChainSync/Server.hs
Comment (nfrisby) [inline ouroboros-consensus/src/Ouroboros/Consensus/MiniProtocol/ChainSync/Server.hs:69]: Would you move this comment sentence to `chainSyncServerForFollower`? It'd be good to put the two driving examples there motivating why `b` does not equal `blk`.

File: ouroboros-consensus/src/Ouroboros/Consensus/MiniProtocol/ChainSync/Server.hs
Comment (nfrisby) [inline ouroboros-consensus/src/Ouroboros/Consensus/MiniProtocol/ChainSync/Server.hs:49]: Would you move this comment sentence to `chainSyncServerForFollower`? It'd be good to put the two driving examples there motivating why `b` does not equal `blk`.

**Comment by nfrisby** [comment]
@EncodePanda would you fixup the title? I just moments ago added you as a reviewer to PR #2903, which elaborates on this sort of meta-level concern; you can find some motivation/details there.

File: ouroboros-consensus/src/Ouroboros/Consensus/Network/NodeToNode.hs
Comment (nfrisby) [inline ouroboros-consensus/src/Ouroboros/Consensus/Network/NodeToNode.hs:117]: Marcin's PR 2090 introduced this argument to all Consensus mini protocol threads in commit 68d35f453c1b1bc1a6eb5ebc38f1bf8392f2be48.

I was looking at what the state was then, but then I realized I should just look at today's `master`. Turns out, none of our mini protocol threads (ie "handlers") use the version information.

That's based on inspection of this code and the definitions it calls to
https://github.com/input-output-hk/ouroboros-network/blob/ed24495e525931306a30de5827ea91caafbec528/ouroboros-consensus/src/Ouroboros/Consensus/Network/NodeToNode.hs#L171-L207

It looks like the NTC analogs don't even have the types for depending on the
version
https://github.com/input-output-hk/ouroboros-network/blob/ed24495e525931306a30de5827ea91caafbec528/ouroboros-consensus/src/Ouroboros/Consensus/Network/NodeToClient.hs#L88-L98

Comment (nfrisby) [reply ouroboros-consensus/src/Ouroboros/Consensus/Network/NodeToNode.hs:117]: So my thought process is: let's remove c6764f7b6c11d020c7bc160ecf4bb31ac6dec7e4 from this PR, open a separate Issue, and ping Marcin on it. Without hearing his thoughts yet, my intuition is [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it) says we should remove the version from all the NTN handlers.

**Review by nfrisby — CHANGES_REQUESTED** [review]
Good catches! I'm happy you're inspecting so thoroughly.

I'm Requesting Changes b/c of the PR title and the separation of concerns (regarding removing the version argument).

File: ouroboros-consensus/src/Ouroboros/Consensus/Network/NodeToNode.hs
Comment (EncodePanda) [reply ouroboros-consensus/src/Ouroboros/Consensus/Network/NodeToNode.hs:117]: I have similar intuition. But if we remove `c6764f7b6c11d020c7bc160ecf4bb31ac6dec7e4` we will also have to remove `9388a549cb0ac01caf4fe643fd022e949c546556` - if we keep the version in `chainSyncHeadersServer` then it simply makes no sense of removing it in the next PR, because `chainSyncHeadersServer` gives nice facade for its usage in `Handlers`.

Maybe it's worth having the discussion about the version numbers being removed from `Handelres` (where they are not needed) now - in this PR?

Comment (EncodePanda) [reply ouroboros-consensus/src/Ouroboros/Consensus/Network/NodeToNode.hs:117]: follow up after meeting with @nfrisby - we will keep this commit but we will not remove version argument from `Handlers`

**Comment by nfrisby** [comment]
@EncodePanda FYI, the CI caught an unused import.

**Comment by nfrisby** [comment]
Did you want to remove the arg from `chainSyncClient` too?

Oof, I forgot that the `TxSub` handlers are defined in the Network Layer. I saw your Slack conversation with Marcin -- it's not clear to me if that's the kind of diff that will incur the merge conflicts you two had discussed.

A few options come to mind:

1. we defer this PR until his `p2p` merge happens
1. we narrow this PR to just the consensus files
1. we get his Approval on your diff (which might also end up being one of the first two options here)

**Comment by EncodePanda** [comment]
@nfrisby I've removed last commit, the one that removed `chainSyncHeadersServer` and `chainSyncBlockServer`. I've created a ticket #2942 (btw a low hanging fruit for newcommer) so that we won't forget to do it once `p2p-master` is in.

I've left changes in `ouroboros-consensus/src/Ouroboros/Consensus/Network/NodeToNode.hs`, those should not block @coot when he eventually merges `p2p-master`

Please do final review of this PR so we can get it in :)

**Comment by EncodePanda** [comment]
@nfrisby let's wrap this up, this way or the other :)

**Review by nfrisby — APPROVED** [review]
Reviewed over Meet 👍

**Pull request merged**
